Marlin/src/inc/Conditionals_post.h

```diff
@@ -2166,7 +2166,7 @@
 #if PIN_EXISTS(Z4_MAX)
   #define HAS_Z4_MAX 1
 #endif
-#if BOTH(HAS_BED_PROBE, HAS_CUSTOM_PROBE_PIN) && PIN_EXISTS(Z_MIN_PROBE)
+#if ENABLED(HAS_BED_PROBE) && PIN_EXISTS(Z_MIN_PROBE)
```

Comment:
Note that the following clause in `Conditionals_LCD.h` is naive. We don't know whether there is a `Z_MIN_PROBE_PIN` defined yet, so this merely signals that one is required.
```cpp
  #if DISABLED(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN) && !BOTH(DELTA, SENSORLESS_PROBING)
    #define HAS_CUSTOM_PROBE_PIN 1
  #endif
```